### ISSUE TITLE
Turn on "noImplicitAny" and "strictPropertyInitializiation" for `fast-foundation`. 

### ISSUE THREAD

**Issue by chrisdholt** [microsoft/fast#5338]
As part of #5303 we turned off @typescript-eslint/typedef globally as it was overwritten across multiple packages. As part of that, we turned on noImplicitAny and strictPropertyInitialization for all packages. Turning this on revealed several issues related to both tsconfig settings in `fast-foundation` and so it was turned off explicitly for that package. The errors should be fixed and addressed as part of this issue and the code should be tested to ensure that there are no breaks or side effects of the change. This issue exists to track turning that back on and resolving the errors.

**Comment by m4thieulavoie** [comment]
@chrisdholt you can assign me this one too 👍

**Comment by m4thieulavoie** [comment]
@chrisdholt https://github.com/microsoft/fast/pull/5370 is ready for review :+1:

**Comment by rajsite** [comment]
An issue we keep hitting when switching from native to FAST Foundation-based equivalents is that the default value for properties are not assigned, like for [text area readOnly](https://github.com/microsoft/fast/blob/155f020427092209ded8feba18d445f6fb1b1877/packages/web-components/fast-foundation/src/text-area/text-area.ts#L23).

Manifests as:
```ts
const fastTextAreaEl = document.createElement(
  'fast-text-area'
) as unknown as TextArea;
document.body.insertAdjacentHTML(
  'beforeend',
  `<div>fast-text-area readOnly default: ${fastTextAreaEl.readOnly}</div>`
); // fast-text-area readOnly default: undefined

const textAreaEl = document.createElement('textarea');
document.body.insertAdjacentHTML(
  'beforeend',
  `<div>textarea readOnly default: ${textAreaEl.readOnly}</div>`
); // textarea readOnly default: false
```
[stackblitz](https://stackblitz.com/edit/fast-default-values?file=index.ts)

It sounds like resolving this issue will clean that up but let me know if it should be a separate issue. 👍

**Comment by chrisdholt** [comment]
Closing this as I don't think we are going to be able to overcome issues with certain attributes which inherit from HTMLElement and don't support undefined values. In these cases, we'd ideally want to either set them to undefined or have them optional, but they essentially are neither. An example was, `id`. 

```ts
public id?: string;
```

The above isn't doable since `id` is simply typed as a string. As a compromise we've set all boolean attributes to have default values. with #6023.
